spark/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala

```diff
@@ -635,7 +636,8 @@ class DeltaLog private(
    */
   protected def checkRequiredConfigurations(): Unit = {
     if (spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_REQUIRED_SPARK_CONFS_CHECK)) {
-      if (!spark.conf.contains(SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key)) {
+      val catalogImpl = spark.conf.get(SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key, null)
```

Comment:
Needed because of https://github.com/apache/spark/commit/6fc176f4f34d73d6f6975836951562243343ba9a.

I am probably going to fix this in spark/master.